modules/assim.sequential/R/Remote_helpers.R

```diff
@@ -0,0 +1,384 @@
+  settings$rundir <- paste0(settings$host$folder,"/", folder_name,"//run")
+  settings$modeloutdir  <-paste0(settings$host$folder,"/", folder_name,"//out")
+  
+  save.setting.dir <- tempdir()
+  PEcAn.settings::write.settings(settings, basename(settingPath), save.setting.dir)
+  
+  # copying over the settings
+  remote.copy.to(
+    my_host,
+    file.path(save.setting.dir, basename(settingPath)),
+    settings$outdir,
+    delete = FALSE,
+    stderr = FALSE
+  )
+  #----------------------------------------------------------------
+  # Copying over the luncher and sending the command
+  #---------------------------------------------------------------
+  # copying over the luncher
+  remote.copy.to(
+    my_host,
+    system.file("RemoteLauncher", "SDA_launcher.R", package = "PEcAn.assim.sequential"),
+    settings$outdir,
+    delete = FALSE,
+    stderr = FALSE
+  )
+  
+  cmd <- paste0("nohup  Rscript ",
+                settings$outdir,"//SDA_launcher.R ", # remote luncher
+                settings$outdir,"//",basename(settingPath), # path to settings
+                " Obs//", basename(ObsPath), # Path to Obs
+                " > ", 
+                settings$outdir,"/SDA_remote_nohup.out 2>&1 &"
+  )
+  
+
+
+   PEcAn.logger::logger.info("Running this command on your remote: \n")
+   PEcAn.logger::logger.info(cmd)
+  
+   
+   #calling SDA
+   out<-remote.execute.R(paste0("system(\" ",cmd, "\")"),
+                         my_host,
+                         user = my_host$user,
+                         scratchdir = ".")
+   
+   
+   
+   # Let's see what is the PID of the job doing the nohup
+   # I'll use this to track the progress of my SDA job
+   PIDS<-remote.execute.cmd(my_host, cmd = "lsof",
+                      args = c(paste0(settings$outdir,"/SDA_remote_nohup.out")))
+   
+   if (length(PIDS)>1){
+     #some cleaning
+     PID<-PIDS[-1] %>% 
+       map_dbl(function(line){
+         ll <- strsplit(line, " ")[[1]]
+         ll <- ll[nchar(ll)>0]
+         ll[2] %>% 
+           as.numeric()
+       }) %>%
+       unique()
+   }else{
+     PEcAn.logger::logger.severe("Something broke the run before it starts!")
+     #TODO: read the nohup.out if it exists
+   }
+
+   #This where you can find your SDA
+   return(list(Remote.Path = settings$outdir,
+               PID = PID))
+  
+}
+
+#' Remote_Sync_launcher
+#'
+#' @param settingPath Path to your local setting .
+#' @param remote.path Path generated by SDA_remote_launcher which shows the path to your remote SDA run.
+#' @param PID PID generated by SDA_remote_launcher which shows the active PID running your SDA job.
+#'
+#' @return
+#' @export
+#' @example 
+#' \dontrun{
+#'  library(PEcAn.all)
+#'  library(purrr)
+#'  
+#'  settingPath <-
+#'    "/fs/data3/hamzed/Projects/GeoTunnel/RemoteSDA/pecan.SDA.4sites.xml"
+#'  ObsPath <-
+#'    "/fs/data3/hamzed/Projects/GEF_MultiSite/Obs/LandTrendr_AGB_output50s.RData"
```

Comment:
less specific paths.  Also, the function documentation doesn't explain what this function actually does